test/datahike/test/transact_test.cljc

```diff
@@ -346,15 +350,96 @@
+      (is (= (:tx-meta tx-report)
+             generated))))
+  (testing "manual txInstant is the same as auto-generated"
+    (let [conn (du/setup-db)
+          date (tools/get-time)
+          tx   (d/transact conn {:tx-data [{:name "Sergey"
+                                            :age  5}]
+                                 :tx-meta {:db/txInstant date}})]
+      (is (= [[1 :age 5 536870913 true]
+              [1 :name "Sergey" 536870913 true]
+              [536870913
+               :db/txInstant
+               date
+               536870913
+               true]]
+             (mapv (comp #(into [] %) seq)
+                   (d/datoms @conn :eavt))))))
+  (testing "missing schema definition"
+    (let [schema [{:db/ident       :name
+                   :db/cardinality :db.cardinality/one
+                   :db/index       true
+                   :db/unique      :db.unique/identity
+                   :db/valueType   :db.type/string}
+                  {:db/ident       :age
+                   :db/cardinality :db.cardinality/one
+                   :db/valueType   :db.type/long}]
+          conn (du/setup-db {:initial-tx schema
+                             :schema-flexibility :write})]
+      (is (thrown-msg? "Bad entity attribute :foo at [:db/add 536870914 :foo :bar 536870914], not defined in current schema"
+                       (d/transact conn {:tx-data [{:name "Sergey"
+                                                    :age  5}]
+                                         :tx-meta {:foo :bar}})))))
+  (testing "meta-data is available on the indices"
+    (let [conn (du/setup-db)
+          tx   (d/transact conn {:tx-data [{:name "Sergey"
+                                            :age  5}]
+                                 :tx-meta {:foo :bar}})]
+      (is (= #{[536870913 :bar]}
+             (d/q '[:find ?e ?v
+                    :where [?e :foo ?v]]
+                  @conn)))))
+  (testing "retracting metadata"
+    (let [conn (du/setup-db)
+          _    (d/transact conn {:tx-data [{:name "Sergey"
+                                            :age  5}]
+                                 :tx-meta {:foo :bar}})
+          _    (d/transact conn {:tx-data [[:db/retract 536870913 :foo :bar]]})]
+      (is (= #{}
+             (d/q '[:find ?e ?v
+                    :where [?e :foo ?v]]
+                  @conn)))))
+  (testing "overwrite metadata"
+    (let [conn (du/setup-db)
+          _    (d/transact conn {:tx-data [{:name "Sergey"
+                                            :age  5}]
+                                 :tx-meta {:foo :bar}})
+          _    (d/transact conn {:tx-data [{:db/id 536870913 :foo :baz}]})]
+      (is (= #{[536870913 :baz]}
+             (d/q '[:find ?e ?v
+                    :where [?e :foo ?v]]
+                  @conn)))))
+  (testing "metadata has txInstant"
+    (let [conn (du/setup-db)
+          _    (d/transact conn {:tx-data [{:name "Sergey"
```

Comment:
You could also check by getting the `:current-tx` out of the TxReport and fetching the meta entity with `pull` and tx ID.